client/verta/tests/unit_tests/test_runtime.py

```diff
@@ -0,0 +1,201 @@
+    will stay thread-safe.
+    """
+    def log_in_context(value):
+        with runtime.context() as test_ctx:
+            runtime.log('test_key', value)
+            time.sleep(random.uniform(0, 1))
+        assert runtime._get_thread_logs() == {}
+        assert test_ctx.logs() == {'test_key': value}
+
+    with futures.ThreadPoolExecutor(max_workers=10) as executor:
+        list(
+            executor.map(
+                log_in_context,
+                [{f'thread_{x}_key': f'thread_{x}_val'} for x in range(101)]
+            )
+        )
+
+
+class TestThreadLocalFunctions(unittest.TestCase):
+    """ Unit tests for the functions that handle thread local variables """
+
+    def test_init_thread_logs(self):
+        """
+        The logs stored in the thread local variable are blank on init.
+        """
+        local_dict: Dict[str, Any] = runtime._THREAD.__dict__
+        assert runtime._get_thread_logs() == {}
+
+    def test_set_thread_logs(self):
+        runtime._set_thread_logs({'test_set': 'test_set'})
+        assert runtime._THREAD.logs ==  {'test_set': 'test_set'}
+
+    def test_get_thread_logs(self):
+        runtime._set_thread_logs({'test_get': 'test_get'})
+        assert runtime._get_thread_logs() ==  {'test_get': 'test_get'}
+        runtime._set_thread_logs({})
+
+    def test_init_thread_validate(self):
+        """
+        The logs stored in the thread local variable are blank on init.
+        """
+        assert runtime._get_validate_flag() == False
+
+    def test_set_thread_logs(self):
+        runtime._set_validate_flag(True)
+        assert runtime._THREAD.validate == True
+
+    def test_get_thread_logs(self):
+        runtime._set_validate_flag(True)
+        assert runtime._get_validate_flag() == True
+        runtime._set_validate_flag(False)
+
+
+class TestContext(unittest.TestCase):
+    """ Unit tests for the verta.runtime.context (context manager) class. """
+
+    def test_logs_are_clean_on_entry(self):
+        """ Thread local var for logs is a blank dict when entering context. """
+        with runtime.context() as ctx:
+            assert ctx.logs() == {}
+
+    def test_logs_are_clean_on_exit(self):
+        """ Thread local var for logs is a blank dict after exiting context. """
+        with runtime.context() as ctx:
+            runtime.log('fake_log', 'fake_value')
+        assert runtime._get_thread_logs() == {}  # outside the context
+
+    def test_final_log_entry_instance_variable(self):
+        """ The instance of the context manager maintains the final log entry in logs(). """
+        with runtime.context() as ctx:
+            runtime.log('fake_log', 'fake_value')
+        assert ctx.logs() == {'fake_log': 'fake_value'}  # outside the context
+
+    def test_validate_flag_true(self):
+        """ The validate argument triggers json validation as expected. """
+        with pytest.raises(TypeError):
+            with runtime.context(validate=True):
```

Comment:
`pytest.raises()` should be as specific as possible in its body. I don't think this test is intended to capture the possibility of `runtime.context(validate=True)` raising a `TypeError`.

```suggestion
        with runtime.context(validate=True):
            with pytest.raises(TypeError):
```

Reply:
Implemented